core/src/main/java/bisq/core/api/CoreDisputeAgentsService.java

```diff
@@ -0,0 +1,145 @@
+    private final P2PService p2PService;
+
+    @Inject
+    public CoreDisputeAgentsService(Config config,
+                                    KeyRing keyRing,
+                                    MediatorManager mediatorManager,
+                                    RefundAgentManager refundAgentManager,
+                                    P2PService p2PService) {
+        this.config = config;
+        this.keyRing = keyRing;
+        this.mediatorManager = mediatorManager;
+        this.refundAgentManager = refundAgentManager;
+        this.p2PService = p2PService;
+    }
+
+    public void registerDisputeAgent(String disputeAgentType, String registrationKey) {
+        if (!p2PService.isBootstrapped())
+            throw new IllegalStateException("p2p service is not bootstrapped yet");
+
+        if (config.baseCurrencyNetwork.isMainnet()
+                || config.baseCurrencyNetwork.isDaoBetaNet()
+                || !config.useLocalhostForP2P)
+            throw new IllegalStateException("dispute agents must be registered in a Bisq UI");
+
+        if (!registrationKey.equals(DEV_PRIVILEGE_PRIV_KEY))
+            throw new IllegalArgumentException("invalid registration key");
+
+        switch (disputeAgentType) {
+            case "arbitrator":
+                throw new IllegalArgumentException("arbitrators must be registered in a Bisq UI");
+            case "mediator":
+            case "refundagent":
+                NodeAddress nodeAddress = new NodeAddress(
+                        InetAddress.getLoopbackAddress().getHostAddress(), config.nodePort);
+                List<String> languageCodes = Arrays.asList("de", "en", "es", "fr");
+                ECKey ecKey = mediatorManager.getRegistrationKey(registrationKey);
+                String signature = mediatorManager.signStorageSignaturePubKey(Objects.requireNonNull(ecKey));
+
+                if (disputeAgentType.equals("mediator"))
+                    registerMediator(nodeAddress, languageCodes, ecKey, signature);
+                else
+                    registerRefundAgent(nodeAddress, languageCodes, ecKey, signature);
+
+                return;
+            default:
+                throw new IllegalArgumentException("unknown dispute agent type " + disputeAgentType);
+        }
+    }
+
+    private void registerMediator(NodeAddress nodeAddress,
+                                  List<String> languageCodes,
+                                  ECKey ecKey,
+                                  String signature) {
+        Mediator mediator = new Mediator(
+                nodeAddress,
+                keyRing.getPubKeyRing(),
+                languageCodes,
+                new Date().getTime(),
+                ecKey.getPubKey(),
+                signature,
+                null,
+                null,
+                null
+        );
+        mediatorManager.addDisputeAgent(mediator, () -> {
+        }, errorMessage -> {
```

Comment:
This is some ugly indentation, seems it's what we have specified as our standard though, and codacy agrees.